src/value.jl

```diff
@@ -80,48 +80,39 @@ Base.promote_rule(::Type{C1}, ::Type{C2}) where {C1<:CatValue, C2<:CatValue} =
 
 Base.convert(::Type{Ref}, x::CatValue) = RefValue{leveltype(x)}(x)
 Base.convert(::Type{String}, x::CatValue) = convert(String, get(x))
-Base.convert(::Type{Any}, x::CatValue) = x
 
 # Defined separately to avoid ambiguities
 Base.convert(::Type{AbstractString}, x::CategoricalString) = x
 Base.convert(::Type{T}, x::T) where {T <: CatValue} = x
 Base.convert(::Type{Union{T, Missing}}, x::T) where {T <: CatValue} = x
 Base.convert(::Type{Union{T, Nothing}}, x::T) where {T <: CatValue} = x
+
+@static if isdefined(Base, :unwrap)
+    Base.unwrap(x::CatValue) = get(x)
+else
 # General fallbacks
-Base.convert(::Type{S}, x::T) where {S, T <: CatValue} =
-    T <: S ? x : convert(S, get(x))
```

Comment:
Just to be sure, do things like `convert(CategoricalValue, x::CategoricalValue{String,UInt32})` work with this PR? That was handled by the `T<:S` branch.